littletable.py

```diff
@@ -2252,6 +2261,94 @@ def tsv_import(
             **kwargs,
         )
 
+    def _excel_import(
+        self,
+        excel_source: _ImportExportDataContainer,
+        transforms: Dict = None,
+        filters: Dict = None,
+        row_class: type = None,
+        limit: int = None,
+        **kwargs,
+    ):
+        if openpyxl is None:
+            raise Exception("openpyxl module not installed")
+
+        def excel_as_dict(filename, **reader_args) -> List[Dict[str, str]]:
+            with closing(openpyxl.load_workbook(filename, read_only=True)) as wb:
+                # read requested sheet if provided on kwargs, otherwise read active sheet
+                requested_sheet = reader_args.get("sheet")
+                ws = wb[requested_sheet] if requested_sheet else wb.active
+                # check whether to include or omit the header
+                header = reader_args.get("fieldnames") or [cell.value for cell in ws[1]]
+                start_position = 0 if reader_args.get("fieldnames") else 1
+
+                # return read data as List[Dict]
+                data = list()
+                for row in list(ws.rows)[start_position:]:
+                    values = {}
+                    for key, cell in zip(header, row):
+                        values[key] = cell.value
+                    data.append(values)
+            return data
```

Comment:
Another stab at excel_as_dict, this time using iterators and making it a generator, so that we minimize the number of copies of the data.

        def excel_as_dict(filename, **reader_args) -> Iterable[Dict[str, Any]]:
            # lazy import openpyxl
            from openpyxl import load_workbook

            with closing(load_workbook(filename, read_only=True)) as wb:
                # read requested sheet if provided on kwargs, otherwise read active sheet
                requested_sheet = reader_args.get("sheet")
                ws = wb[requested_sheet] if requested_sheet else wb.active

                rows_iter = iter(ws.rows)

                # check whether to include or omit the header
                header = (reader_args.get("fieldnames")
                          or [cell.value for cell in next(rows_iter)])

                for row in rows_iter:
                    yield {key: cell.value for key, cell in zip(header, row)}

Reply:
Did the changes in https://github.com/ptmcg/littletable/pull/4/commits/58eb3f9a11d7e6672a2bc7295186b09558460f86